### ISSUE TITLE
NVENC Hardware Acceleration

### ISSUE THREAD

**Issue by kmcclive** [kmcclive/tricycle#72]
As a user, I want to use Nvidia's technology to accelerate transcodes so that jobs take much less time to complete.

**Comment by palaashatri** [comment]
Is someone working on this, or should I give it a try?

**Comment by kmcclive** [comment]
No, no one is working on it.  I don't know if it's fully possible at this time.  The problem is the HDR support.  Tricycle currently uses X265 parameters to specify the HDR metadata, which writes it to the video stream.  I opened #70 to generalize this beyond X265.  HDR10, the HDR standard that Tricycle supports, also allows specifying the HDR metadata in the container.  However, ffmpeg did not have a way of doing this last time I looked.  I tried submitting a patch to add the support but it was rejected.  I considered distributing a custom build of ffmpeg with Tricycle, but had difficulty getting it to build on all platforms.  You are welcome to give it a try, but the implementation should support HDR with both MKV and MP4.  Since Tricycle is targeted at uneducated users, I try to avoid format-specific features.
